konfera/models/schedule.py

```diff
@@ -23,3 +25,35 @@ class Schedule(KonferaModel):
 
     def __str__(self):
         return _('%(start)s (%(duration)s min)') % {'start': self.start, 'duration': self.duration}
+
+    def clean(self, *args, **kwargs):
+        # Only approved talk can be scheduled
+        if self.talk.status != 'approved':
+            raise ValidationError(_('You cannot schedule unapproved talks'))
+
+        # Event is related to location and location has room,
+        # make sure selected room in schedule belongs to Event's location
+        room = get_model('Room').objects.filter(location=event.location)
+        if location_room.first() != self.room:
+            raise ValidationError(_('The room that you have chosen is '
+                'not part of the scheduled room'))
+
+        # Make sure date and time is within the event's range
+
+        # Make sure system does not allow store two events at the same
+        # time in the same room, eg. schedule datetime + duration in room is unique.
+        schedules = get_model('Schedule').objects.filter(
+            start=self.start, duration=self.duration, room=self.room)
+        if schedules.exists():
```

Comment:
What about when they are not fully overlapping?

For example:
schedule1 starts at 10:00 with duration 60minutes
schedule2 starts at 10:30 with duration 30minutes